File: src/1528-Shuffle-String/shuffle_string.py
from typing import List

class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        l: List[str] = [""] * len(indices)
        i: int = 0

        for c in s:
            l[indices[i]] = c
            i += 1
        
        ans: str = "".join(l)
        
        return ans

    
if __name__ == "__main__":
    s: str =  "codeleet"
    indices: List[int] = [4,5,6,7,0,2,1,3]
    solution = Solution()
    
    print(solution.restoreString(s, indices))

File: src/1528-Shuffle-String/shuffle_string.cpp
#include <iostream>
#include <string>
#include <vector>

class Solution {
public:
    std::string restoreString(std::string s, std::vector<int>& indices) {
        std::string ans(s.length(), ' '); 

        for (int i = 0; i < indices.size(); ++i) {
            ans[indices[i]] = s[i];
        }

        return ans;
    }
};

int main() {
    std::string s = "codeleet";
    std::vector<int> indices = {4, 5, 6, 7, 0, 2, 1, 3};
    Solution solution;

    std::cout << solution.restoreString(s, indices);
    
    return 0;
}